---
title: "reuterin_deseq2"
author: "Ryan Rebernick"
date: "4/23/2021"
output: html_document
---

# Description: 
This performs DE analysis using DESeq2 on Reuterin and control samples as well as QC. Several comparisons are done with the ultimate decision being to remove sample s121113. This was a difficult decision as PCA plot of all the samples shows that s121112 appears to be the outlier. However after plotting the z-scaled expression of all of the genes it is apparent that s121113 appears to be the sample that is the most different. This is confirmed by plotting all samples on PCA for just the DEGs for each comparison. In each one, s121113 is the most different from all the other samples and is spread across PC1. Uses GRCm38.p6

Also generates RNA-seq heatmaps used in figures.


# Import Packages
```{r, include=F}

library('DESeq2')

library('plyr')
library('dplyr')
library('magrittr')
library('data.table')

library('ComplexHeatmap')
#install.packages("viridis")
library(viridis)
library(ggfortify)
library(cowplot)


```



# Load ensembl to common name conversion from ensembl biomart (GRCm38.p6)
```{r}

# Load and clean
ensembl <- fread('~/Desktop/projects/reuterin/data/ensembl/Grcm38p6_geneNames.txt')
ensembl %<>% dplyr::select(`Gene stable ID version`, `Gene name`)
colnames(ensembl) <- c('gene', 'geneName')

```



# Load count matrix and clean for analysis
Half of these samples correspond to a different experiment. Select the ones needed.
```{r, warning=F, message=F}

# Load
raw <- fread('~/Desktop/projects/reuterin/data/rnaseq/counts.txt', data.table = F)

# rename/reorganize data frame for deseq 
colnames(raw) <- c('Ensembl', 'chr', 'start', 'end', 'strand', 'length', 's121096', 's121097', 's121098',
                   's121099','s121100', 's121101', 's121103', 's121104', 's121105', 's121106', 's121107',
                   's121108', 's121109', 's121110', 's121111', 's121112', 's121113', 's122108')
nms <- raw$Ensembl
## select needed cols and reorder
raw %<>% dplyr::select(s121105, s121106, s121107,
                s121111, s121112, s121113)
rownames(raw) <- nms

```



# Comparison of CONTROL and REUTERIN for all samples 


### Prepare for DESEQ2 differential expression
```{r}

# select only the samples that passed QC (see multiqc_report.html) and rename columns
## All passed

# create matrix with sample info
sampleInfo <- as.data.frame(matrix(c('s121105', 's121106', 's121107',
                's121111', 's121112', 's121113', 
                   c(replicate(3, 'ctrl'), replicate(3, 'reuterin'))), ncol=2))
colnames(sampleInfo) <- c('sample', 'condition')
rownames(sampleInfo) <- c('s121105', 's121106', 's121107',
                's121111', 's121112', 's121113')
# Factor variable
sampleInfo$condition <- factor(sampleInfo$condition, levels = c('ctrl', 'reuterin'))

# check names align
all(rownames(sampleInfo) == colnames(raw))

# create a DESeq data object from count matrix
dds <- DESeqDataSetFromMatrix(countData = raw,
                                   colData = sampleInfo,
                                   design = ~ condition)


```

### Transformation and PCA
Consider removing s121112 as outlier per PCA
```{r}

#Transform
vsd <- vst(dds)

#PCA
plotPCA(vsd) + theme_cowplot(12)

#PCA - change labeling
plotPCA(vsd, intgroup = c('sample', 'condition')) + theme_cowplot(12)


```

### Run differential expression
```{r}

res <- DESeq(dds)
res <- results(res, independentFiltering = T, contrast=c("condition","reuterin","ctrl"))

# order results by smallest P-value and summarize
resOrdered <- res[order(res$pvalue),]
summary(resOrdered)


```

### Volcano plot
```{r}
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))
  with(subset(res, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
  with(subset(res, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```

### Export data
```{r}

# make data frame
reuterin <- as.data.frame(res@listData, row.names = res@rownames)
reuterin$gene <- rownames(reuterin)
head(reuterin)

# Add common names
reuterin <- merge(reuterin, ensembl, all.x = T, by = 'gene')

# Reformat
reuterin %<>% dplyr::select(geneName, gene, baseMean:padj) %>%
  arrange(padj) 
head(reuterin)

# Add counts data
reuterin.cts <- merge(reuterin, raw, by.x = 'gene', by.y=0)

# save output
# write.table(reuterin, '~/Desktop/projects/reuterin/out/deseq2/reuterin_allSamps_deseq2.txt', row.names = F, col.names = T, sep = '\t', quote = F)

```

### Clean data for heatmap
```{r}

toplot <- reuterin.cts %>%
  dplyr::filter(padj < 0.1)
nms <- toplot$geneName
toplot %<>%
  dplyr::select(s121105:s121113) %>%
  t() %>%
  scale(center = T, scale = T) %>%
  t() %>%
  set_rownames(nms)


```

### draw heatmap using all 6 samples
```{r}

Heatmap(toplot,
        show_row_dend = F,
        show_column_dend = F,
        show_row_names = F,
        show_column_names = T,
        cluster_rows = T,
        cluster_columns = F,
        row_km = 2,
        row_km_repeats = 100,
        column_order = 1:6,
        column_split = factor(
          c(rep('Control', 3), rep('Reuterin', 3))),
        width = unit(6, "cm"),
        height = unit(7, "cm"),
        column_title_gp = gpar(fontsize=6),
        heatmap_legend_param = list(title = "Z-score"))

```

### PCA plot of just DEGs for all samples
```{r}

toPca <- reuterin.cts %>%
  dplyr::filter(padj < 0.1)
toPca %<>%
  dplyr::select(s121105:s121113) %>%
  t() %>%
  as.data.frame()
toPca$group <- rownames(toPca)

pca_res <- prcomp(toPca %>% dplyr::select(-group), scale. = T)

autoplot(object = pca_res, 
                  data = toPca,
                  colour = 'group',
                  size = 6) + 
  theme_cowplot(12)

```



# Remove sample s121112 and compare between CONTROL and REUTERIN for 5 samples

### Remove sample and recreate deseq object for pca
```{r}

# remove from matrix file
raw.f <- raw %>% dplyr::select(-s121112)

# remove from metadata
sampleInfo.f <- sampleInfo %>% dplyr::filter(sample != 's121112')

# ensure rows/cols still equal
all(rownames(sampleInfo.f) == colnames(raw.f))

# create a DESeq data object from count matrix
dds <- DESeqDataSetFromMatrix(countData = raw.f,
                                   colData = sampleInfo.f,
                                   design = ~ condition)
```


### Transformation and PCA
```{r}

#Transform
vsd <- vst(dds)

#PCA
plotPCA(vsd) + theme_cowplot(12)

#PCA - change labellign
plotPCA(vsd, intgroup = c('sample', 'condition')) + theme_cowplot(12)


```


### Run differential expression
```{r}

res <- DESeq(dds)
res <- results(res, independentFiltering = T, contrast=c("condition","reuterin","ctrl"))

# order results by smallest P-value and summarize
resOrdered <- res[order(res$pvalue),]
summary(resOrdered)


```

### Volcano plot
```{r}
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))
  with(subset(res, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
  with(subset(res, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```


### Export data
```{r}

# make data frame
reuterin <- as.data.frame(res@listData, row.names = res@rownames)
reuterin$gene <- rownames(reuterin)
head(reuterin)

# Add common names
reuterin <- merge(reuterin, ensembl, all.x = T, by = 'gene')

# Reformat
reuterin %<>% dplyr::select(geneName, gene, baseMean:padj) %>%
  arrange(padj) 
head(reuterin)

# Add counts data
reuterin.cts <- merge(reuterin, raw, by.x = 'gene', by.y=0)

# save output
#write.table(reuterin, '~/Desktop/projects/reuterin/out/deseq2/reuterin_deseq2.txt', row.names = F, col.names = T, sep = '\t', quote = F)

```


### Clean data for heatmap
```{r}

toplot <- reuterin.cts %>%
  dplyr::filter(padj < 0.1)
nms <- toplot$geneName
toplot %<>%
  dplyr::select(s121105:s121113) %>%
  dplyr::select(-s121112) %>%
  t() %>%
  scale(center = T, scale = T) %>%
  t() %>%
  set_rownames(nms)


```

### draw heatmap
```{r}

Heatmap(toplot,
        col = viridis(100),
        show_row_dend = F,
        show_column_dend = F,
        show_row_names = F,
        show_column_names = T,
        cluster_rows = T,
        cluster_columns = F,
        row_km = 2,
        row_km_repeats = 100,
        column_order = 1:5,
        column_split = factor(
          c(rep('Control', 3), rep('Reuterin', 2))),
        width = unit(6, "cm"),
        height = unit(7, "cm"),
        column_title_gp = gpar(fontsize=6),
        heatmap_legend_param = list(title = "Z-score"))

```

### PCA plot of just DEGs for all samples
```{r}

toPca <- reuterin.cts %>%
  dplyr::filter(padj < 0.1)
toPca %<>%
  dplyr::select(s121105:s121113) %>%
  t() %>%
  as.data.frame()
toPca$group <- rownames(toPca)

pca_res <- prcomp(toPca %>% dplyr::select(-group), scale. = T)

autoplot(object = pca_res, 
                  data = toPca,
                  colour = 'group',
                  size = 6) + 
  theme_cowplot(12)

```



# Remove sample s121113 and compare between CONTROL and REUTERIN for 5 samples


### Remove sample and recreate deseq object for pca
```{r}

# remove from matrix file
raw.f <- raw %>% dplyr::select(-s121113)

# remove from metadata
sampleInfo.f <- sampleInfo %>% dplyr::filter(sample != 's121113')

# ensure rows/cols still equal
all(rownames(sampleInfo.f) == colnames(raw.f))

# create a DESeq data object from count matrix
dds <- DESeqDataSetFromMatrix(countData = raw.f,
                                   colData = sampleInfo.f,
                                   design = ~ condition)
```


### Transformation and PCA
```{r}

#Transform
vsd <- vst(dds)

#PCA
plotPCA(vsd) + theme_cowplot(12)

#PCA - change labellign
plotPCA(vsd, intgroup = c('sample', 'condition')) + theme_cowplot(12)

```

#### Save pca plot
```{r}

# with labels
pdf(paste0("~/Desktop/projects/reuterin/out/figures/rna_seq_pca_wo_s121113_",format(Sys.time(), "%Y-%m-%d"), ".pdf"),height = 4, width = 4)
plotPCA(vsd) + 
  geom_point(size=4) +
  theme_cowplot(12) +
  scale_color_viridis(discrete = T, end = 0.5) +
  ylim(c(-7,3))
dev.off()

# without labels
pdf(paste0("~/Desktop/projects/reuterin/out/figures/rna_seq_pca_wo_s121113_noLab_",format(Sys.time(), "%Y-%m-%d"), ".pdf"),height = 3, width = 3)
plotPCA(vsd) + 
  geom_point(size=4) +
  theme_cowplot(12) +
  scale_color_viridis(discrete = T, end = 0.5) +
  ylim(c(-7,3)) +
  xlab(NULL) + 
  ylab(NULL) +
  theme(legend.text = element_blank(),
      legend.title = element_blank(),
      axis.text = element_blank())
dev.off()

```



### Run differential expression
```{r}

res <- DESeq(dds)
res <- results(res, independentFiltering = T, contrast=c("condition","reuterin","ctrl"))

# order results by smallest P-value and summarize
resOrdered <- res[order(res$pvalue),]
summary(resOrdered)


```

### Volcano plot
```{r}
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))
  with(subset(res, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
  with(subset(res, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```


### Export data
```{r}

# make data frame
reuterin <- as.data.frame(res@listData, row.names = res@rownames)
reuterin$gene <- rownames(reuterin)
head(reuterin)

# Add common names
reuterin <- merge(reuterin, ensembl, all.x = T, by = 'gene')

# Reformat
reuterin %<>% dplyr::select(geneName, gene, baseMean:padj) %>%
  arrange(padj) 
head(reuterin)

# Add counts data
reuterin.cts <- merge(reuterin, raw, by.x = 'gene', by.y=0)

# save output
#write.table(reuterin, '~/Desktop/projects/reuterin/out/deseq2/reuterin_deseq2.txt', row.names = F, col.names = T, sep = '\t', quote = F)
#write.table(reuterin.cts, '~/Desktop/projects/reuterin/int/deseq2/reuterin_deseq2_wcounts.txt', row.names = F, col.names = T, sep = '\t', quote = F)


```


### Clean data for heatmap
```{r}

toplot <- reuterin.cts %>%
  dplyr::filter(padj < 0.1)
nms <- toplot$geneName
toplot %<>%
  dplyr::select(s121105:s121112) %>%
  t() %>%
  scale(center = T, scale = T) %>%
  t() %>%
  set_rownames(nms)


```

### draw heatmap
```{r}

Heatmap(toplot,
        show_row_dend = F,
        show_column_dend = F,
        show_row_names = F,
        show_column_names = T,
        cluster_rows = T,
        cluster_columns = F,
        row_km = 2,
        row_km_repeats = 100,
        column_order = 1:5,
        column_split = factor(
          c(rep('Control', 3), rep('Reuterin', 2))),
        width = unit(6, "cm"),
        height = unit(7, "cm"),
        column_title_gp = gpar(fontsize=6),
        heatmap_legend_param = list(title = "Z-score")
        )

# Flip
h <- Heatmap(t(toplot),
        show_row_dend = F,
        show_column_dend = F,
        show_row_names = F,
        show_column_names = F,
        cluster_rows = F,
        cluster_columns = T,
        column_km = 2,
        column_km_repeats = 100,
        row_order = 1:5,
        row_split = factor(
          c(rep('Control', 3), rep('Reuterin', 2))),
        width = unit(7, "cm"),
        height = unit(2.5, "cm"),
        row_title_gp = gpar(fontsize=6),
        column_title = NULL,
        heatmap_legend_param = list(title = "Z-score"))

h
#pdf(paste0('~/Desktop/projects/reuterin/out/figures/reuterin_degs_heatmap_', format(Sys.time(), "%Y-%m-%d"), '.pdf'), height = 4, width = 6)
h
dev.off()


# Flip w/o labels
h <- Heatmap(t(toplot),
        show_row_dend = F,
        show_column_dend = F,
        show_row_names = F,
        show_column_names = F,
        cluster_rows = F,
        cluster_columns = T,
        column_km = 2,
        column_km_repeats = 100,
        row_order = 1:5,
        row_split = factor(
          c(rep('Control', 3), rep('Reuterin', 2))),
        width = unit(7, "cm"),
        height = unit(2.5, "cm"),
        row_title_gp = gpar(fontsize=6),
        column_title = NULL,
        heatmap_legend_param = list(title = NULL),
        row_title = NULL)

h
#pdf(paste0('~/Desktop/projects/reuterin/out/figures/reuterin_degs_heatmap_noLab_', format(Sys.time(), "%Y-%m-%d"), '.pdf'), height = 4, width = 6)
h
dev.off()

```

### PCA plot of just DEGs for all samples
```{r}

toPca <- reuterin.cts %>%
  dplyr::filter(padj < 0.1)
toPca %<>%
  dplyr::select(s121105:s121113) %>%
  t() %>%
  as.data.frame()
toPca$group <- rownames(toPca)

pca_res <- prcomp(toPca %>% dplyr::select(-group), scale. = T)

autoplot(object = pca_res, 
                  data = toPca,
                  colour = 'group',
                  size = 6) + 
  theme_cowplot(12)

```

### PCA plot of just DEGs for just the 5 included samples (excl s121113)
```{r}

toPca <- reuterin.cts %>%
  dplyr::filter(padj < 0.1)
toPca %<>%
  dplyr::select(s121105:s121112) %>%
  t() %>%
  as.data.frame()
toPca$group <- rownames(toPca)

pca_res <- prcomp(toPca %>% dplyr::select(-group), scale. = T)

autoplot(object = pca_res, 
                  data = toPca,
                  colour = 'group',
                  size = 6) + 
  theme_cowplot(12)

```