---
title: "reuterin_deseq2"
author: "Ryan Rebernick"
date: "4/23/2021"
output: html_document
---

# Setup for analysis with DESEQ2


####Import Packages
```{r, include=F}

library('DESeq2')

library('plyr')
library('dplyr')
library('magrittr')
library('data.table')

```

#### Load ensembl to common name conversion from ensembl biomart (GRCm38.p6)
```{r}

# Load and clean
ensembl <- fread('~/Desktop/projects/reuterin/data/ensembl/Grcm38p6_geneNames.txt')
ensembl %<>% dplyr::select(`Gene stable ID version`, `Gene name`)
colnames(ensembl) <- c('gene', 'geneName')

```

####Load count matrix and clean for analysis
**Half of these samples are rat and correspond to a different experiment.
```{r, warning=F, message=F}

# Load
raw <- fread('~/Desktop/projects/reuterin/data/rnaseq/counts.txt', data.table = F)

# rename/reorganize data frame for deseq 
colnames(raw) <- c('Ensembl', 'chr', 'start', 'end', 'strand', 'length', 's121096', 's121097', 's121098',
                   's121099','s121100', 's121101', 's121103', 's121104', 's121105', 's121106', 's121107',
                   's121108', 's121109', 's121110', 's121111', 's121112', 's121113', 's122108')
nms <- raw$Ensembl
## select needed cols and reorder
raw %<>% dplyr::select(s121105, s121106, s121107,
                s121111, s121112, s121113)
rownames(raw) <- nms

```



# Comparison of all samples - PCA


####Prepare for DESEQ2 differential expression
```{r}

# select only the samples that passed QC (see multiqc_report.html) and rename columns
## All passed

# create matrix with sample info
sampleInfo <- as.data.frame(matrix(c('s121105', 's121106', 's121107',
                's121111', 's121112', 's121113', 
                   c(replicate(3, 'ctrl'), replicate(3, 'reuterin'))), ncol=2))
colnames(sampleInfo) <- c('sample', 'condition')
rownames(sampleInfo) <- c('s121105', 's121106', 's121107',
                's121111', 's121112', 's121113')
# Factor variable
sampleInfo$condition <- factor(sampleInfo$condition, levels = c('ctrl', 'reuterin'))

# check names align
all(rownames(sampleInfo) == colnames(raw))

# create a DESeq data object from count matrix
dds <- DESeqDataSetFromMatrix(countData = raw,
                                   colData = sampleInfo,
                                   design = ~ condition)


```


####Transformation and PCA
Consider removing s121112 as outlier per PCA
```{r}

#Transform
vsd <- vst(dds)

#PCA
plotPCA(vsd)

#PCA - change labellign
plotPCA(vsd, intgroup = c('sample', 'condition'))


```

# Remove sample s121112

#### Remove sample and recreate deseq object for pca
```{r}

# remove from matrix file
raw %<>% dplyr::select(-s121112)

# remove from metadata
sampleInfo %<>% dplyr::filter(sample != 's121112')

# ensure rows/cols still equal
all(rownames(sampleInfo) == colnames(raw))

# create a DESeq data object from count matrix
dds <- DESeqDataSetFromMatrix(countData = raw,
                                   colData = sampleInfo,
                                   design = ~ condition)
```


####Transformation and PCA
```{r}

#Transform
vsd <- vst(dds)

#PCA
plotPCA(vsd)

#PCA - change labellign
plotPCA(vsd, intgroup = c('sample', 'condition'))


```


# Comparison between CONTROL and REUTERIN

#### Run differential expression
```{r}

res <- DESeq(dds)
res <- results(res, independentFiltering = T, contrast=c("condition","reuterin","ctrl"))

# order results by smallest P-value and summarize
resOrdered <- res[order(res$pvalue),]
summary(resOrdered)


```

####Volcano plot
```{r}
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))
  with(subset(res, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
  with(subset(res, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```


####Export data
```{r}

# make data frame
reuterin <- as.data.frame(res@listData, row.names = res@rownames)
reuterin$gene <- rownames(reuterin)
head(reuterin)

# Add common names
reuterin <- merge(reuterin, ensembl, all.x = T, by = 'gene')

# Reformat
reuterin %<>% select(geneName, gene, baseMean:padj) %>%
  arrange(padj) 
head(reuterin)

# Add counts data
reuterin.cts <- merge(reuterin, raw, by.x = 'gene', by.y=0)

# save output
write.table(reuterin, '~/Desktop/projects/reuterin/out/deseq2/reuterin_deseq2.txt', row.names = F, col.names = T, sep = '\t', quote = F)

```

